salt/modules/boto_ec2.py

```diff
@@ -1758,20 +1758,18 @@ def set_volumes_tags(tag_maps, authoritative=False, dry_run=False,
     '''
     ret = {'success': True, 'comment': '', 'changes': {}}
     running_states = ('pending', 'rebooting', 'running', 'stopping', 'stopped')
+    ### First creeate a dictionary mapping all changes for a given volume to it's volume ID...
+    tag_sets = {}
     for tm in tag_maps:
-        filters = tm.get('filters')
-        tags = tm.get('tags')
-        if not isinstance(filters, dict):
-            raise SaltInvocationError('Tag filters must be a dictionary: got {0}'.format(filters))
-        if not isinstance(tags, dict):
-            raise SaltInvocationError('Tags must be a dictionary: got {0}'.format(tags))
+        filters = dict(tm.get('filters', {}))
+        tags = dict(tm.get('tags', {}))
         args = {'return_objs': True, 'region': region, 'key': key, 'keyid': keyid, 'profile': profile}
         new_filters = {}
         log.debug('got filters: {0}'.format(filters))
         instance_id = None
         in_states = tm.get('in_states', running_states)
         try:
-            for k, v in six.iteritems(filters):
+            for k, v in filters.items():
```

Comment:
Hey Ryan, in general I only use six in places where it's needed (e.g. where "standard" python can't do the same job directly).  In this case, and for dict iterating in general, I habitually use dict.items() -- simply because it works on python 2 and python 3 directly without any need for the compatibility layer.

The size of 'filters' is guaranteed to be tiny, so iterating in RAM (on python 2) will be no hit, which would be the only driver for insisting on six.iteritems() I can think of.